lib/rpc/src/eth_call_handler.rs

```diff
@@ -407,7 +443,8 @@ impl<RpcStorage: ReadRpcStorage> EthCallHandler<RpcStorage> {
                 }
                 Err(
                     InvalidTransaction::CallGasCostMoreThanGasLimit
-                    | InvalidTransaction::OutOfGasDuringValidation,
+                    | InvalidTransaction::OutOfGasDuringValidation
+                    | InvalidTransaction::OutOfNativeResourcesDuringValidation,
```

Comment:
this method is now over 200 LoC. Others are over 100 in this file.
Not necessarily in this PR, but  maybe makes sense to refactor this module? At the very least extract some functions, but I think some new local abstractions can do well here.